### PR TITLE
Use a separate sauce labs token per committer

### LINKED ISSUE #10086: Per-committer Sauce Tokens
We might be able to alleviate the SauceLabs strain by using per-committer (at least per-core-committer) Sauce tokens.

When it's a PR build, based on who commits, we use their tokens to run integration tests. When it's a branch build, we can use the official tokens, etc.

### PR DESCRIPTION
**Pull request by rsimha-amp**
Fixes #10086

### CODE REVIEW

**Comment by rsimha-amp** [comment]
This PR introduces per-user sauce labs credentials. (I'm still testing it to make sure that we can initially connect using `amphtml`'s credentials and then reconnect using per-user creds.)

I have a hunch we'll need to move away from JWT to store the `amphtml` saucelab credentials, and encode and check them in to `sauce-credentials.json`. @cramforce, @erwinmombay, @jridgewell: Let me know your initial thoughts before I update this PR with obfuscated `amphtml` sauce labs credentials.

File: build-system/sauce-credentials.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "_comment": "Encoded per-user Sauce labs credentials. Add yours to this file.",
+  "credentials": {
+    "rsimha@ampproject.org": {
+      "sauce_username": "rsimha",
```
Comment (jridgewell) [inline build-system/sauce-credentials.json]: Repetitive `sauce_` is repetitive.

File: build-system/pr-check.js [hunk 2]
```diff
@@ -27,13 +27,15 @@
 const atob = require('atob');
 const exec = require('./exec.js').exec;
 const execOrDie = require('./exec.js').execOrDie;
+const fs = require('fs-extra');
```
Comment (jridgewell) [inline build-system/pr-check.js]: Why `fs-extra`?

Comment (rsimha-amp) [inline build-system/pr-check.js]: Fixed.

File: build-system/sauce-credentials.json (hunk 1 above)
Comment (rsimha-amp) [inline build-system/sauce-credentials.json]: Fixed.

**Comment by rsimha-amp** [comment]
Update: In order to connect to sauce labs with the right set of credentials, we're going to have to move from using JWT to store encrypted sauce credentials, to using the mechanism we use for Percy, where we store encoded credentials in a json file and pick the one to use at runtime. Trying to reconnect with a new set of credentials after having connected with amphtml credentials does not work. See https://travis-ci.org/ampproject/amphtml/jobs/247269899

Lazily connecting to sauce when we know we're going to run integration tests has the added benefit of reducing the connection load on sauce servers. However, this won't be as secure as with using JWT. (We currently do this for Percy.)

@cramforce, @erwinmombay, WDYT?

**Comment by rsimha-amp** [comment]
Lazily connecting to sauce labs in `pr-check.js` won't be possible, since the sauce connection needs to be kept alive in a background process and cleanly torn down after testing. This is all handled by the [sauce_connect addon](https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/addons/sauce_connect.rb), and it makes little sense to replicate that code in js.

I've moved the username / access key extraction logic to a separate script that I'm trying to run between when JWT is initialized (thereby setting the encrypted amphtml sauce credentials), and when the sauce connection is established. So if this works as I think it does, we'll insert per-user credentials into the env vars in between the two steps.

**Comment by rsimha-amp** [comment]
Yay, at long last, this worked. Merging after checks complete.

**Pull request merged**